server/vitals.js

```diff
@@ -149,22 +149,59 @@ function sendClientButtonStatusChanges(clientButtonStatusChanges) {
   // Loop through each client to create the Twilio messages
   Object.values(clientButtonStatusChanges).forEach(({ client, disconnectedButtons, reconnectedButtons }) => {
     let message = [i18next.t('buttonStatusChangeStart', { lng: client.language, clientDisplayName: client.displayName })]
+    let messageBilingual = null
+
+    // check for bilingual client language
+    if (message[0].split('\n---\n').length > 1) {
+      const split = message[0].split('\n---\n')
+
+      message = [split[0]]
+      messageBilingual = [split[1]]
+    }
 
     if (disconnectedButtons.length > 0) {
       const buttonDisplayNames = disconnectedButtons.sort().join(', ') // sorted alphabetically
-      message.push(i18next.t('buttonStatusChangeDisconnected', { lng: client.language, buttonDisplayNames }))
+      let disconnectButtonsMessage = i18next.t('buttonStatusChangeDisconnected', { lng: client.language, buttonDisplayNames })
+
+      // handle bilingual client language translation
+      if (messageBilingual !== null) {
+        const split = disconnectButtonsMessage.split('\n---\n')
+
+        disconnectButtonsMessage = split[0]
+        messageBilingual.push(split[1])
+      }
+
+      message.push(disconnectButtonsMessage)
     }
 
     if (reconnectedButtons.length > 0) {
       const buttonDisplayNames = reconnectedButtons.sort().join(', ') // sorted alphabetically
-      message.push(i18next.t('buttonStatusChangeReconnected', { lng: client.language, buttonDisplayNames }))
+      let reconnectedButtonsMessage = i18next.t('buttonStatusChangeReconnected', { lng: client.language, buttonDisplayNames })
+
+      // handle bilingual client language translation
+      if (messageBilingual !== null) {
+        const split = reconnectedButtonsMessage.split('\n---\n')
+
+        reconnectedButtonsMessage = split[0]
+        messageBilingual.push(split[1])
+      }
+
+      message.push(reconnectedButtonsMessage)
     }
 
     // join the message parts with spaces
     message = message.join(' ')
 
-    // send SMS text messages to each of the client's heartbeat phone numbers
-    client.heartbeatPhoneNumbers.forEach(phoneNumber => {
+    // handle bilingual client language translation
```

Comment:
Cool. I see what you're doing here. I think you're right that this produces a more readable text message than what I suggested above, so I'm going to strikeout my earlier comment. 

My main concern with doing it the way that you have is the split on `'\n---\n'`. This could be something that someone changes in the `.json` language file without realizing that it has an implication in the code.

What do you think about having three different messages for the case where there are reconnections, where there are disconnections, and where there both disconnections and reconnections as the three strings in the `.json` language files? So instead of 
```
  "buttonStatusChangeDisconnected": "Les Buttons suivants ont été déconnectés: {{ buttonDisplayNames }}.\n---\nThe following buttons have disconnected: {{ buttonDisplayNames }}.",
  "buttonStatusChangeReconnected": "Les Buttons suivants ont été reconnectés: {{ buttonDisplayNames }}.\n---\nThe following buttons have reconnected: {{ buttonDisplayNames }}.",
  "buttonStatusChangeStart": "Des modifications de connexion ont été apportées aux Buttons de {{ clientDisplayName }}.\n---\nThere has been connection changes for the buttons at {{ clientDisplayName }}.",
```
we instead did something like:
```
  "buttonStatusChangeDisconnected": "Des modifications de connexion ont été apportées aux Buttons de {{ clientDisplayName }}.\n\nLes Buttons suivants ont été déconnectés: \n{{ disconnectedButtonDisplayNames }}.\n---\nThere has been connection changes for the buttons at {{ clientDisplayName }}.\n\nThe following buttons have disconnected:\n{{ disconnectedButtonDisplayNames }}.",
  "buttonStatusChangeReconnected": "Des modifications de connexion ont été apportées aux Buttons de {{ clientDisplayName }}.\n\nLes Buttons suivants ont été reconnectés:\n{{ reconnectedButtonDisplayNames }}.\n---\nThere has been connection changes for the buttons at {{ clientDisplayName }}.\n\nThe following buttons have reconnected: \n{{ reconnectedButtonDisplayNames }}.",
  "buttonStatusChangeDisconnectedAndReconnected": "Des modifications de connexion ont été apportées aux Buttons de {{ clientDisplayName }}.\n\nLes Buttons suivants ont été déconnectés:\n{{ disconnectedButtonDisplayNames }}.\n\nLes Buttons suivants ont été reconnectés:\n{{ reconnectedButtonDisplayNames }}.\n---\nThere has been connection changes for the buttons at {{ clientDisplayName }}.\n\nThe following buttons have disconnected:\n{{ disconnectedButtonDisplayNames }}.\n\nThe following buttons have reconnected:\n{{ reconnectedButtonDisplayNames }}.",
```
then you could pick which string based on the size of the `disconnectedButtons` and `reconnectedButtons` arrays and it would be same logic for uni-lingual and bi-lingual clients.

Reply:
I like this idea! I'll implement it.

Reply:
Done!